Read from IRequest instead of reading twice

LukasReschke opened the pull request:
Potentially fixes https://github.com/owncloud/core/issues/14541 and https://github.com/owncloud/core/issues/14506

… let's see what our Jenkins says to this …

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c316509f-1d5a-405a-8199-15f94a33c835) was created.

LukasReschke commented:
I removed the TemplateLayout tests that I added ages ago since they were horribly anyways and such stuff really prevents from having a sane Request class…

LukasReschke commented:
Other apps that needs fixing: https://github.com/search?utf8=%E2%9C%93&q=user%3Aowncloud+file_get_contents%28%22php%3A%2F%2Finput%22%29&type=Code&ref=searchresults

*sigh*

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c316509f-1d5a-405a-8199-15f94a33c835) completed: **No new issues**

Raydiation commented:
Good job, looks good :+1:

bantu commented on lib/private/api.php:
```diff
@@ -84,11 +84,13 @@ public static function register($method, $url, $action, $app,
 	 * @param array $parameters
 	 */
 	public static function call($parameters) {
+		$request = \OC::$server->getRequest();
```
`$method = OC::$server->getRequest()getMethod();` ?

LukasReschke replied:
Mhm… Would require me to instantiate another one for `getParams`. This way it's IMHO easier to migrate it in the future to a testable controller as we can just replace `$request` with a member variable.

bantu replied:
Oh err yeah. Could use `$method = $request->getMethod();`.

LukasReschke replied:
So true :) :rocket:

LukasReschke replied:
Done.

LukasReschke commented:
Requires test execution on 5.4 and Co. – as discussed with @DeepDiver1975 yesterday. Will ping him again on monday.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f7fa0dd5-e883-42ac-90a1-99256fb35b1e) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f7fa0dd5-e883-42ac-90a1-99256fb35b1e) completed: **12 new issues, 2 updated code elements**

LukasReschke commented:
:bell: :bell: :bell:  … pinging @DeepDiver1975 again as promised on Friday …  :bell: :bell: :bell:

DeepDiver1975 commented:
@dragotin  @rullzer can I ask you to test this PR with respect to the failing PUT via OCS API? THX

rullzer commented:
My tests all work as expected again on php 5.5.22! :+1:

DeepDiver1975 commented:
:+1:
